src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2177,6 +2190,12 @@ private unsafe ParsingError PrivateParseMinimal()
                     if (err != ParsingError.None)
                         return err;
 
+                    // Unix paths must start with '/'
+                    if (InFact(Flags.UnixPath) && idx < (ushort)length && pUriString[idx] != '/')
```

Comment:
InFact(Flags.UnixPath) will only ever return true if we're on a unix system, right (since otherwise it wouldn't have been recognized)?  If so, we may want to preface this check with ```!IsWindowsSystem &&```... that way, assuming my previous comment about Roslyn is correct, this whole if statement would end up being compiled away on Windows and would only be in the Unix binary.